src/portable/portable.c

```diff
@@ -706,6 +707,14 @@ static int install_chroot_dropin(
 
                         return -ENOMEM;
 
+                r = load_os_release_pairs_container_prefix_env("/", &os_release_pairs);
```

Comment:
hmm, i'd use a different prefix here. i.e. "portable_host_" or so? after all portable services and containers are different things.

Yes, this means clients need to test for both variables, but that should be acceptable, no?

This matters in particular when you nest stuff, i.e. have a portable service inside a container

Reply:
On one hand, my concern with this would be that applications then need to be aware of the "containerization" system in use (with a very liberal use of the term container of course, as portable services are not technically speaking traditional containers, if there is such a thing). Say Flatpak added this interface, would it be "flatpak_host_foo" ?

On the other hand, you raise a very valid and important point about nesting, that I missed as it didn't affect what I'm doing - speaking in terms of generic interfaces, what should the behavior be? For our use case and in my mind, this was about knowing "the thing I'm running in" and "the thing that runs everything on this machine". If nesting is introduced, this doesn't work - foo_host_ is from the second level.

Should the behavior perhaps be "passthrough"? Eg, if nspawn.c if we have container_host_* in our env, then we are about to start a nested guest, so instead of parsing from scratch just copy the variables from the existing env?

Reply:
you have a point, we should probably use names that indicate the level of sandboxing/containerization instead of the implementation.

But I think it does make sense to distinguish two levels here: i.e. full-machine containerization as in "more like VMs", and "per-service containerization" as in "just sandbox this one service". it's a blurry line of course, as docker as one example is mostly doing hte latter but can also be told to do the former. And nspawn typically does the former, but can also be used to do the latter.

As so far we used the container_xyz env var prefix only for the first type of container I think we should stick to it too. But for the second type we should maybe use "service_host_" as prefix, which would leave the door open for flatpak to use the same, as I think portable services are much more like flatpak in this case.

Does that make sense?

Reply:
Yes, that makes sense. So in nspawn, if start mode is START_PID1 or START_PID2 I'd use service_host_, else container_host_ - did I understand correctly? And in portabled always use service_host_

Reply:
hmm, didn't think about the --as-pid2 thing... but you are probably right.

Reply:
Ok, thanks.

What about the previous point on nesting? As it is right now, with a situation like:

host -> nspawn -> portable service

the service would see service_host_FOO with values from the nspawn instance, not the host.

Should I check the env, and if there are existing variables pass them through directly instead of reading new ones from the middle layer?

Reply:
(pushed update with service_host/container_host)

Reply:
> But for the second type we should maybe use "service_host_" as prefix, which would leave the door open for flatpak to use the same

Flatpak is application-oriented: the things in the container are very much apps, not services, which makes this prefix seem a little odd for the Flatpak case.

Reply:
I guess the difficulty is finding a good name that can be shared by portable services, Flatpak, Snaps, and single-process mode Docker/nspawn. Any suggestion?